### DISCUSSION TITLE
Current hardening measures for UEFI binaries

### DISCUSSION THREAD

**Discussion by RaitoBezarius** [uapi-group/specifications#61] [General]
Hello there,

I am working in NixOS to bring native UEFI cross-compilation (using LLVM lld-link), we got somewhere really nice, but we discovered that our default hardening measures do not really apply (obviously, ELF are different): read-only relocations, bindnow, etc.

Does anyone know if there's any plan to support better hardening measures for such binaries? This seems like an easy thing to get and important one.

Thank you!

**Comment by poettering** [comment]
last time a looked uefi had no shared library concept. hence relocations/bindnow simply don't apply.

**Reply by RaitoBezarius** [reply]
Interesting, but my question is open on also: stackprotector, fortify3, format, strictoverflow.

**Reply by RaitoBezarius** [reply]
(also: https://doc.rust-lang.org/rustc/exploit-mitigations.html)

**Reply by bluca** [reply]
The NX bit is the only hardening I'm aware of, and it's going to be required by firmwares "soon" (TM)

**Reply by RaitoBezarius** [reply]
Which is the default for MSVC linker, right? `/NXCOMPAT`
